scripts/build_test.py

```diff
@@ -36,7 +107,197 @@ def test_insert_hash(self):
                           'path/to/file.min.fedcba.js')
         # pylint: enable=protected-access
 
+    def test_ensure_directory_exists(self):
+        """Test ensure_directory_exists to make sure non-existent parent
+        directory of random.js must be created after calling function.
+        """
+        # Setup a sandbox folder to assert for non-existence.
```

Comment:
I would prefer having no tests for the parts of the code which modify the os / filesystem. I think that is fine.

Or, perhaps better, you can use self.swap() to swap out the filesystem bits (e.g. mock shutil.copyfile, shutil.rmtree) and just assert that the mock function was called. See what we do for e.g. some of the tests in core.platform.search.

Reply:
I agreed. Will get to this test after PR is stable.